api/src/db/migrations/2024.06.07T00.36.14.clean-user-group-names.ts

```diff
@@ -1,12 +1,16 @@
 import type { Migration } from "@/db/umzug"
-import { UserGroup } from "@/models"
 
-export const up: Migration = async () => {
-  await UserGroup.findEach(async (userGroup) => {
-    await userGroup.update({
-      name: userGroup.name.trim().replace(/\s+/g, " "),
-    })
-  })
+export const up: Migration = async ({ context: queryInterface }) => {
+  await queryInterface.sequelize.transaction(async (transaction) => {
+    const [rows]: any[] = await queryInterface.sequelize.query('SELECT name from user_groups', { transaction });
+    for (const row of rows) {
+      const trimmedName = row.name.trim().replace(/\s+/g, " ")
+      await queryInterface.sequelize.query('UPDATE user_groups SET name = ? WHERE id = ?', {
+        replacements: [trimmedName, row.id],
+        transaction
+      });
```

Comment:
I don't recommend using a transaction here; you are better off doing the changes incrementally as this could overflow RAM or lock the database when run in production. Production servers usually have a lot fewer resources than a personal machine.

Reply:
Generally, the migration library should handle transactionality for you, (though with JavaScript you might have to hack it in manually at the config level). In this case it's hacked in manually at the config level; see `api/src/db/utils/sequelize-auto-transaction-resolver.ts`